### ISSUE TITLE
Is this a legitimate way to do saliency mapping with MolGraphConv-featurized molecules?

### ISSUE THREAD

**Issue by jadeshi** [deepchem/deepchem#3215]
Hi,
Wanted to ask the experts here for some feedback, since I'm still fairly new at using graph-based neural networks. 

Say I train a binary classifier GCNModel using a set of MolGraphConv-featurized compounds to predict whether a compound will have some property, and I'd like to get an idea of which atoms contribute most strongly to the prediction. Currently what I'm doing to probe this is to iterate through every atom of the featurized molecule and set the corresponding row in node_features for that atom to 0, and then using my trained model to predict using that modified molecule with the zeroed out atom. I repeat this across all atoms and treat the magnitudes of the difference between the new probability and the original as the "saliency map" for the molecule. I figure this should closely mimic the gradient of the output with respect to that atom's features if I were to do saliency mapping properly by backpropagating, which does not seen straightforward with this type of model. Is this an acceptable way to go about it, or is there a better way?

Thanks!

**Comment by rbharath** [comment]
I think this would be a pretty reasonable way to compute an approximate
saliency map. I haven't tried it myself so can't say for sure though

**Comment by jadeshi** [comment]
Thanks Bharath! Good to know what I'm proposing is not completely crazy. I did some very simple benchmarking using a matched pair with a one-atom difference and it seemed to be able to highlight the that local neighborhood as contributing most strongly. I think I'll stick with this until something better pops up. Thanks again!
